### PR TITLE
Initial content

### PR DESCRIPTION
**Pull request by rafaparadela**
This PR creates a Giter8 template that includes a multimodule project:

- `protocol` where is described a single protocol `PeopleService` with a single operation `getPerson` with Freestyle-RPC annotations (serialized with Protobuf).
- `server` that runs the RPC server and includes an interpreter for `PeopleService`.
- `client` that consumes the service exposed by the server.

### CODE REVIEW

File: src/main/g8/project/ProjectPlugin.scala [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  private lazy val logSettings: Seq[Def.Setting[_]] = Seq(
+    libraryDependencies ++= Seq(
+      "ch.qos.logback"    % "logback-classic" % V.logbackClassic,
+      "io.chrisdavenport" %% "log4cats-core"  % V.log4cats,
+      "io.chrisdavenport" %% "log4cats-slf4j" % V.log4cats
+    ))
+
+
+  lazy val rpcProtocolSettings: Seq[Def.Setting[_]] = Seq(
+    libraryDependencies ++= Seq("io.frees" %% "frees-rpc-client-core" % V.freestyleRPC)
```
Comment (AdrianRaFo) [inline src/main/g8/project/ProjectPlugin.scala]: you need to add here `idlType := "proto",` if not it will be serialized to `avro` by default

**Review by AdrianRaFo — APPROVED** [review]

File: src/main/g8/project/ProjectPlugin.scala (hunk 1 above)
Comment (rafaparadela) [reply src/main/g8/project/ProjectPlugin.scala]: I'm defining the service in Scala manually as:

```scala
@service(Protobuf) trait PeopleService[F[_]] {

  def getPerson(request: GetPersonRequest): F[GetPersonResponse]

}
```

I'm not interested in generating the proto fields. Actually I'm removing the `sbt-frees-rpc-idlgen` plugin.

File: project/build.properties [hunk 2]
```diff
@@ -0,0 +1 @@
+sbt.version=1.1.0
```
Comment (juanpedromoreno) [inline project/build.properties]: 1.2.1

File: project/giter8.sbt [hunk 3]
```diff
@@ -0,0 +1 @@
+addSbtPlugin("org.foundweekends.giter8" %% "sbt-giter8" % "0.11.0-M2")
```
Comment (juanpedromoreno) [inline project/giter8.sbt]: This can be `0.11.0`

File: src/main/g8/build.sbt [hunk 4]
```diff
@@ -0,0 +1,17 @@
+import ProjectPlugin._
+
+lazy val protocol = project in file("protocol") settings rpcProtocolSettings
```
Comment (juanpedromoreno) [inline src/main/g8/build.sbt]: Personally, and based on the current standards in the frees-io projects we are not using the infix syntax to express the sbt settings.

`lazy val protocol = project.in(file("protocol")).settings(rpcProtocolSettings)`

File: src/main/g8/client/src/main/scala/ClientApp.scala [hunk 5]
```diff
@@ -0,0 +1,26 @@
...
+
+  def main(args: Array[String]): Unit = {
+    (for {
+      logger <- Stream.eval(Slf4jLogger.fromName[IO]("Client"))
+      client <- {
+        implicit val l = logger
+        PeopleServiceClient.createClient[IO]("localhost", 19683)
+      }
+      getPersonResponse <- Stream.eval(client.getPerson("foo")).as(println)
+    } yield (getPersonResponse)).compile.toVector.unsafeRunSync()
```
Comment (juanpedromoreno) [inline src/main/g8/client/src/main/scala/ClientApp.scala]: I feel like we don't need to create an `fs2.Stream` here and then evaluate it immediately. What do you think?

File: src/main/g8/client/src/main/scala/PeopleServiceClient.scala [hunk 6]
```diff
@@ -0,0 +1,56 @@
...
+object PeopleServiceClient {
+
+  def apply[F[_]: Effect](clientF: F[PeopleService.Client[F]])(
+      implicit L: Logger[F]): PeopleServiceClient[F] =
+    new PeopleServiceClient[F] {
+
+      def getPerson(name: String): F[Person] =
+        for {
+          client <- clientF
+          _ <- L.info(s"Request: \$name")
```
Comment (juanpedromoreno) [inline src/main/g8/client/src/main/scala/PeopleServiceClient.scala]: Do you want to escape the `$` character?

File: src/main/g8/project/ProjectPlugin.scala [hunk 7]
```diff
@@ -0,0 +1,70 @@
...
+
+  object autoImport {
+
+    lazy val V = new {
+      val fs2            = "0.10.1"
+      val log4cats       = "0.1.0"
+      val logbackClassic = "1.2.3"
+      val circe          = "0.10.0-M1"
+      val freestyleRPC   = "0.14.0"
+      val http4s         = "0.18.15"
```
Comment (juanpedromoreno) [inline src/main/g8/project/ProjectPlugin.scala]: Are we using http4s?

File: src/main/g8/project/ProjectPlugin.scala [hunk 8]
```diff
@@ -0,0 +1,70 @@
...
+  override def trigger: PluginTrigger = allRequirements
+
+  object autoImport {
+
+    lazy val V = new {
+      val fs2            = "0.10.1"
+      val log4cats       = "0.1.0"
+      val logbackClassic = "1.2.3"
+      val circe          = "0.10.0-M1"
+      val freestyleRPC   = "0.14.0"
```
Comment (juanpedromoreno) [inline src/main/g8/project/ProjectPlugin.scala]: I'd try to use the latest versions of everything when possible

File: src/main/g8/project/build.properties [hunk 9]
```diff
@@ -0,0 +1 @@
+sbt.version = 1.2.0
```
Comment (juanpedromoreno) [inline src/main/g8/project/build.properties]: 1.2.1

File: src/main/g8/project/ProjectPlugin.scala [hunk 10]
```diff
@@ -0,0 +1,70 @@
...
+        "-Yno-adapted-args",
+        "-Ywarn-dead-code",
+        "-Ywarn-numeric-widen",
+        "-Ywarn-value-discard",
+        "-Xfuture",
+        "-Ywarn-unused-import"
+      ),
+      scalafmtCheck := true,
+      scalafmtOnCompile := true,
+      addCompilerPlugin("org.scalamacros" % "paradise" % "2.1.0" cross CrossVersion.full)
```
Comment (juanpedromoreno) [inline src/main/g8/project/ProjectPlugin.scala:78]: I think we could use the 2.1.1 version

File: src/main/g8/server/src/main/scala/PeopleServiceHandler.scala [hunk 11]
```diff
@@ -0,0 +1,15 @@
...
+import cats.effect.Sync
+import cats.syntax.functor._
+import $package$.protocol.services._
+import io.chrisdavenport.log4cats.Logger
+
+class PeopleServiceHandler[F[_]: Sync](implicit L: Logger[F])
+    extends PeopleService[F] {
+
+  override def getPerson(request: GetPersonRequest): F[GetPersonResponse] =
+    L.info(s"PeopleService - Request: \$request")
```
Comment (juanpedromoreno) [inline src/main/g8/server/src/main/scala/PeopleServiceHandler.scala]: escaping `$`?

**Review by juanpedromoreno — CHANGES_REQUESTED** [review]

File: src/main/g8/client/src/main/scala/PeopleServiceClient.scala (hunk 6 above)
Comment (rafaparadela) [reply src/main/g8/client/src/main/scala/PeopleServiceClient.scala]: Yes. This is the code that Giter8 needs to compile, the way how g8 usually detects variables is using `$`, so all the string-interpolation inside of the `g8` needed to be scaped :-(

File: src/main/g8/client/src/main/scala/ClientApp.scala (hunk 5 above)
Comment (rafaparadela) [reply src/main/g8/client/src/main/scala/ClientApp.scala]: Initially (based on the Adri's example) the client boot was a fully StreamApp, but I added some changes but I have kept how the RPC client is created (via `ClientCache`) and this is a constraint at the moment.

File: src/main/g8/client/src/main/scala/PeopleServiceClient.scala (hunk 6 above)
Comment (juanpedromoreno) [reply src/main/g8/client/src/main/scala/PeopleServiceClient.scala]: I see, got it 👍

File: src/main/g8/project/ProjectPlugin.scala (hunk 7 above)
Comment (rafaparadela) [reply src/main/g8/project/ProjectPlugin.scala]: Nope. There are come version that can be dropped off here, thanks.

File: src/main/g8/project/ProjectPlugin.scala (hunk 8 above)
Comment (rafaparadela) [reply src/main/g8/project/ProjectPlugin.scala]:
```scala
      val fs2            = "0.10.4"
      val log4cats       = "0.1.1"
      val logbackClassic = "1.2.3"
      val freestyleRPC   = "0.14.1"
```

File: src/main/g8/server/src/main/scala/PeopleServiceHandler.scala (hunk 11 above)
Comment (rafaparadela) [reply src/main/g8/server/src/main/scala/PeopleServiceHandler.scala]: Giter8 thing

File: src/main/g8/project/ProjectPlugin.scala (hunk 10 above)
Comment (rafaparadela) [reply src/main/g8/project/ProjectPlugin.scala:78]: 👍

File: src/main/g8/project/build.properties (hunk 9 above)
Comment (rafaparadela) [reply src/main/g8/project/build.properties]: 👍

**Review by AdrianRaFo — APPROVED** [review]

**Review by franciscodr — APPROVED** [review]

**Pull request merged**